packages/code-du-travail-frontend/src/outils/DureePreavisRetraite/steps/Anciennete.tsx

```diff
@@ -1,9 +1,21 @@
-import React from "react";
+import React, { useEffect } from "react";
 
 import { TextQuestion } from "../../common/TextQuestion";
+import { WizardStepProps } from "../../common/type/WizardType";
 import { isPositiveNumber } from "../../common/validators";
+import { usePublicodes } from "../../publicodes";
+import { mapToPublicodesSituation } from "../../publicodes/Utils";
+
+function AncienneteStep({ form }: WizardStepProps): JSX.Element {
+  const publicodesContext = usePublicodes();
+
+  useEffect(() => {
```

Comment:
note for later, we could add an event ha,dler on the wizzard to notify step changes to we could recalculate situation before changeing step, or pass a form instance to the publicode context and move the useEffect there

Reply:
Indeed! I think the next step is to create a dedicated wizard for publicodes simulator which can be improved without impacted existing simulator.